src/frequenz/dispatch/_actor_dispatcher.py

```diff
@@ -138,18 +139,21 @@ def __init__(
         self,
         actor_factory: Callable[[DispatchInfo, Receiver[DispatchInfo]], Actor],
         running_status_receiver: Receiver[Dispatch],
+        map_dispatch: Callable[[Dispatch], int],
```

Comment:
IMHO `map_dispatch` is very opaque, I would call this `dispatch_identity` (or maybe even `dispatch_hash`, as this is what we are doing, is a hash, but maybe identity makes the particular intention of the hash here more clear).

I would also allow `None` in here too and treat mapping by `id` as a special (default) case. The identity of a dispatch is the `dispatch.id` unless you want to merge them somehow. Then you can either change the code to deal with `None` or do a `self._dispatch_identity = dispatch_identity or lambda d: d.id` at this level.